### PR TITLE
Clang-tidy and include-what-you-use warnings

### PR DESCRIPTION
**Pull request by mjcarroll**
Will also bump subprocess.h when this lands: https://github.com/sheredom/subprocess.h/pull/75

### CODE REVIEW

File: test/integration/implptr/implptr_test_classes.hh [hunk 1]
```diff
@@ -102,7 +102,7 @@ namespace gz
       public: int GetInt() const;
 
       /// \brief Set the int value held by the pimpl
```
Comment (ahcorde) [inline test/integration/implptr/implptr_test_classes.hh:104]: include `\param _value` here and below

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: test/integration/implptr/implptr_test_classes.hh (hunk 1 above)
Comment (mjcarroll) [reply test/integration/implptr/implptr_test_classes.hh:104]: Done.

**Review by ahcorde — APPROVED** [review]

File: test/integration/implptr/implptr_test_classes.hh [hunk 2]
```diff
@@ -95,19 +103,23 @@ namespace gz
...
+      public: [[nodiscard]] int GetInt() const;
 
       /// \brief Set the int value held by the pimpl
-      public: void SetInt(const int _value);
+      /// \param _value Integer value to set
+      public: void SetInt(int _value);
 
       /// \brief Get the string value held by the pimpl
-      public: const std::string &GetString() const;
+      public: [[nodicard]] const std::string &GetString() const;
```
Comment (iche033) [inline test/integration/implptr/implptr_test_classes.hh:119]:
```suggestion
      public: [[nodiscard]] const std::string &GetString() const;
```

**Comment by azeey** [comment]
DCO is failing

**Pull request merged**